ksqldb-functional-tests/src/test/resources/query-validation-tests/fk-join.json

```diff
@@ -12,6 +12,18 @@
         "message": "Invalid join condition: table-table joins require to join on the primary key of the right input table. Got LEFT_TABLE.ID1 = RIGHT_TABLE.F2."
       }
     },
+    {
+      "name": "Should fail on right non-key attribute for inner-join -- revers join condition order",
```

Comment:
I don't think we need copies of these tests with the feature flag disabled. If the purpose of the test is to check that table-table joins are only allowed on primary keys, then tests for that already exist. If the purpose is to check that the behavior does not change if the join condition is flipped, then that belongs in the regular `joins.json` file since it's not related to foreign key joins.

I think adding duplicates of all the new FK join tests will unnecessarily blow up this file and create more busywork for us (harder reviews, more work when we ungate the feature flag) than it's worth.

Reply:
What is your concrete proposal? I think we should have some of those tests, to make sure that the condition is correctly identified even if the join condition is flipped. And yes, we should have corresponding tests in `joins.json` in addition.

Reply:
My concrete proposal is:
- delete all the new tests in this PR that do not have the feature flag enabled
- remove the comments from the copies of the tests with the feature flag enabled about deleting the test when ungating from the feature flag
- remove `- feature flag enabled` from the names of the copies of the tests that have the feature flag enabled

If we're planning to support flipped join conditions out of the box (which it sounds like we are), then we should add positive test coverage to ensure this is properly supported. I will add that into https://github.com/confluentinc/ksql/pull/7528.

Reply:
I thought about this a little bit. I agree that those this test (and it's "siblings") should not be part of fk-join test. However, `joins.json` is very long and hard to keep an overview what we test and what not. Will add `table-table-join.json` and move these 10 test over. Want to keep the reverse order as I think we should test that the condition is detected for both cases (and also that we "flip" left/right condition in the error message).

I would hope we can cleanup `join.json` eventually, but it's a little tricky to "move" test as we also need to migrate old plans accordingly. Plan tp can dig into this as a follow up.